hoot-services/src/main/java/hoot/services/controllers/grail/WaitOverpassUpdate.java

```diff
@@ -119,17 +120,26 @@ public CommandResult execute() {
                         responseStream = GrailResource.getUrlInputStreamWithNullHostnameVerifier(url);
                     }
 
+                    // get overpass response
                     br = new BufferedReader(new InputStreamReader(responseStream));
-                    String line;
-                    while ((line = br.readLine()) != null) {
-                        if (line.contains(lastId)) {
-                            foundChangesetId = true;
-                            break;
-                        }
-                    }
+                    String urlResult = br.lines().collect(Collectors.joining());
                     br.close();
+                    JSONParser parser = new JSONParser();
+                    JSONObject json = (JSONObject) parser.parse(urlResult);
+                    JSONArray elementsArray = (JSONArray) json.get("elements");
+
+                    // if create then we should have 1 element found with the specified feature type + id
+                    // if delete and no element with specified id is found then we assume it was deleted
+                    if ((operationType.equals("create") && elementsArray.size() == 1) ||
+                            (operationType.equals("delete") && elementsArray.size() == 0)) {
+                        elementFound = true;
+                    } else if (operationType.equals("modify") && elementsArray.size() > 0) {
+                        // For modify check if the version matches
```

Comment:
Should this check that the version is >= to what hoot pushed?  In case something else modified it in the meantime.